src/libcore/ptr.rs

```diff
@@ -2077,6 +2077,32 @@ impl<T: ?Sized> PartialEq for *mut T {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<T: ?Sized> Eq for *mut T {}
 
+#[unstable(feature = "ptr_dangling", issue = "0")]
+/// Returns a const raw pointer that is dangling, but well-aligned and not null.
+///
+/// This is used where a non-null pointer is required.
+///
+/// # Safety
+///
+/// This function is safe to call, but the raw pointer it returns does not
+/// point to any value.
+pub fn dangling<T>() -> *const T {
```

Comment:
Can this function be `const`, just like `null()`?

Reply:
Ah great idea, align_of is const so it should be no problem.